cmd/kms-rest/startcmd/start.go

```diff
@@ -188,3 +209,100 @@ func healthCheckHandler(rw http.ResponseWriter, r *http.Request) {
 		log.Errorf("healthcheck response failure, %s", err)
 	}
 }
+
+type keystoreProvider struct {
+	storageProvider storage.Provider
+	kmsCreator      provider.KMSCreator
+	crypto          crypto.Crypto
+}
+
+func (k keystoreProvider) StorageProvider() storage.Provider {
+	return k.storageProvider
+}
+
+func (k keystoreProvider) KMSCreator() provider.KMSCreator {
+	return k.kmsCreator
+}
+
+func (k keystoreProvider) Crypto() crypto.Crypto {
+	return k.crypto
+}
+
+type kmsProvider struct {
+	storageProvider ariesstorage.Provider
+	secretLock      secretlock.Service
+}
+
+func (k kmsProvider) StorageProvider() ariesstorage.Provider {
+	return k.storageProvider
+}
+
+func (k kmsProvider) SecretLock() secretlock.Service {
+	return k.secretLock
+}
+
+func createKeystoreProvider(parameters *kmsRestParameters) (provider.Provider, error) {
+	c, err := tinkcrypto.New()
+	if err != nil {
+		return nil, err
+	}
+
+	return keystoreProvider{
+		storageProvider: memstore.NewProvider(),
+		kmsCreator:      prepareKMSCreator(parameters),
+		crypto:          c,
+	}, nil
+}
+
+func prepareKMSCreator(_ *kmsRestParameters) provider.KMSCreator {
```

Comment:
Is there a reason for the unused `kmsRestParameters`?

Reply:
indeed, we don't need it now, so removed